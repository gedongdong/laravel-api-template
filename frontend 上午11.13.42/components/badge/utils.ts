import { PresetColorTypes } from '../_util/colors';

export function isPresetColor(color?: string): boolean {
  return (PresetColorTypes as any[]).indexOf(color) !== -1;
}
