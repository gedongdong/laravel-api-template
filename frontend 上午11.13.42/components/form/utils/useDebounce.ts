import type { Ref } from 'vue';
import { shallowRef, watchEffect } from 'vue';

export default function useDebounce<T>(value: Ref<T[]>): Ref<T[]> {
  const cacheValue = shallowRef(value.value.slice());
  let timeout: any = null;
  watchEffect(() => {
    clearTimeout(timeout);
    timeout = setTimeout(
      () => {
        cacheValue.value = value.value;
      },
      value.value.length ? 0 : 10,
    );
  });

  return cacheValue;
}
