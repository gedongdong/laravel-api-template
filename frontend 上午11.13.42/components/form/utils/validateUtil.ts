import RawAsyncValidator from 'async-validator';
import { cloneVNode } from 'vue';
import { warning } from '../../vc-util/warning';

import { setValues } from './valueUtil';
import { defaultValidateMessages } from './messages';
import { isValidElement } from '../../_util/props-util';
import type { InternalNamePath, RuleError, RuleObject, ValidateOptions } from '../interface';

// Remove incorrect original ts define
const AsyncValidator: any = RawAsyncValidator;

/**
 * Replace with template.
 *   `I'm ${name}` + { name: 'bamboo' } = I'm bamboo
 */
function replaceMessage(template: string, kv: Record<string, string>): string {
  return template.replace(/\$\{\w+\}/g, (str: string) => {
    const key = str.slice(2, -1);
    return kv[key];
  });
}

async function validateRule(
  name: string,
  value: any,
  rule: RuleObject,
  options: ValidateOptions,
  messageVariables?: Record<string, string>,
): Promise<string[]> {
  const cloneRule = { ...rule };

  // Bug of `async-validator`
  delete (cloneRule as any).ruleIndex;
  delete (cloneRule as any).trigger;

  // We should special handle array validate
  let subRuleField: RuleObject = null;
  if (cloneRule && cloneRule.type === 'array' && cloneRule.defaultField) {
    subRuleField = cloneRule.defaultField;
    delete cloneRule.defaultField;
  }

  const validator = new AsyncValidator({
    [name]: [cloneRule],
  });

  const messages = setValues({}, defaultValidateMessages, options.validateMessages);
  validator.messages(messages);

  let result = [];

  try {
    await Promise.resolve(validator.validate({ [name]: value }, { ...options }));
  } catch (errObj) {
    if (errObj.errors) {
      result = errObj.errors.map(({ message }, index: number) =>
        // Wrap VueNode with `key`
        isValidElement(message) ? cloneVNode(message, { key: `error_${index}` }) : message,
      );
    } else {
      console.error(errObj);
      result = [(messages.default as () => string)()];
    }
  }

  if (!result.length && subRuleField) {
    const subResults: string[][] = await Promise.all(
      (value as any[]).map((subValue: any, i: number) =>
        validateRule(`${name}.${i}`, subValue, subRuleField, options, messageVariables),
      ),
    );

    return subResults.reduce((prev, errors) => [...prev, ...errors], []);
  }

  // Replace message with variables
  const kv = {
    ...(rule as Record<string, string | number>),
    name,
    enum: (rule.enum || []).join(', '),
    ...messageVariables,
  };

  const fillVariableResult = result.map(error => {
    if (typeof error === 'string') {
      return replaceMessage(error, kv);
    }
    return error;
  });

  return fillVariableResult;
}

/**
 * We use `async-validator` to validate the value.
 * But only check one value in a time to avoid namePath validate issue.
 */
export function validateRules(
  namePath: InternalNamePath,
  value: any,
  rules: RuleObject[],
  options: ValidateOptions,
  validateFirst: boolean | 'parallel',
  messageVariables?: Record<string, string>,
) {
  const name = namePath.join('.');

  // Fill rule with context
  const filledRules: RuleObject[] = rules
    .map((currentRule, ruleIndex) => {
      const originValidatorFunc = currentRule.validator;
      const cloneRule = {
        ...currentRule,
        ruleIndex,
      };

      // Replace validator if needed
      if (originValidatorFunc) {
        cloneRule.validator = (rule: RuleObject, val: any, callback: (error?: string) => void) => {
          let hasPromise = false;

          // Wrap callback only accept when promise not provided
          const wrappedCallback = (...args: string[]) => {
            // Wait a tick to make sure return type is a promise
            Promise.resolve().then(() => {
              warning(
                !hasPromise,
                'Your validator function has already return a promise. `callback` will be ignored.',
              );

              if (!hasPromise) {
                callback(...args);
              }
            });
          };

          // Get promise
          const promise = originValidatorFunc(rule, val, wrappedCallback);
          hasPromise =
            promise && typeof promise.then === 'function' && typeof promise.catch === 'function';

          /**
           * 1. Use promise as the first priority.
           * 2. If promise not exist, use callback with warning instead
           */
          warning(hasPromise, '`callback` is deprecated. Please return a promise instead.');

          if (hasPromise) {
            (promise as Promise<void>)
              .then(() => {
                callback();
              })
              .catch(err => {
                callback(err || ' ');
              });
          }
        };
      }

      return cloneRule;
    })
    .sort(({ warningOnly: w1, ruleIndex: i1 }, { warningOnly: w2, ruleIndex: i2 }) => {
      if (!!w1 === !!w2) {
        // Let keep origin order
        return i1 - i2;
      }

      if (w1) {
        return 1;
      }

      return -1;
    });

  // Do validate rules
  let summaryPromise: Promise<RuleError[]>;

  if (validateFirst === true) {
    // >>>>> Validate by serialization
    summaryPromise = new Promise(async (resolve, reject) => {
      /* eslint-disable no-await-in-loop */
      for (let i = 0; i < filledRules.length; i += 1) {
        const rule = filledRules[i];
        const errors = await validateRule(name, value, rule, options, messageVariables);
        if (errors.length) {
          reject([{ errors, rule }]);
          return;
        }
      }
      /* eslint-enable */

      resolve([]);
    });
  } else {
    // >>>>> Validate by parallel
    const rulePromises: Promise<RuleError>[] = filledRules.map(rule =>
      validateRule(name, value, rule, options, messageVariables).then(errors => ({ errors, rule })),
    );

    summaryPromise = (
      validateFirst ? finishOnFirstFailed(rulePromises) : finishOnAllFailed(rulePromises)
    ).then((errors: RuleError[]): RuleError[] | Promise<RuleError[]> => {
      // Always change to rejection for Field to catch
      return Promise.reject<RuleError[]>(errors);
    });
  }

  // Internal catch error to avoid console error log.
  summaryPromise.catch(e => e);

  return summaryPromise;
}

async function finishOnAllFailed(rulePromises: Promise<RuleError>[]): Promise<RuleError[]> {
  return Promise.all(rulePromises).then(
    (errorsList: RuleError[]): RuleError[] | Promise<RuleError[]> => {
      const errors: RuleError[] = [].concat(...errorsList);

      return errors;
    },
  );
}

async function finishOnFirstFailed(rulePromises: Promise<RuleError>[]): Promise<RuleError[]> {
  let count = 0;

  return new Promise(resolve => {
    rulePromises.forEach(promise => {
      promise.then(ruleError => {
        if (ruleError.errors.length) {
          resolve([ruleError]);
        }

        count += 1;
        if (count === rulePromises.length) {
          resolve([]);
        }
      });
    });
  });
}
