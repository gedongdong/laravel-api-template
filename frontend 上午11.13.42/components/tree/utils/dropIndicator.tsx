import type { CSSProperties } from 'vue';

export const offset = 4;

export default function dropIndicatorRender(props: {
  dropPosition: -1 | 0 | 1;
  dropLevelOffset: number;
  indent: number;
  prefixCls: string;
  direction: 'ltr' | 'rtl';
}) {
  const { dropPosition, dropLevelOffset, prefixCls, indent, direction = 'ltr' } = props;
  const startPosition = direction === 'ltr' ? 'left' : 'right';
  const endPosition = direction === 'ltr' ? 'right' : 'left';
  const style: CSSProperties = {
    [startPosition]: `${-dropLevelOffset * indent + offset}px`,
    [endPosition]: 0,
  };
  switch (dropPosition) {
    case -1:
      style.top = `${-3}px`;
      break;
    case 1:
      style.bottom = `${-3}px`;
      break;
    default:
      // dropPosition === 0
      style.bottom = `${-3}px`;
      style[startPosition] = `${indent + offset}px`;
      break;
  }
  return <div style={style} class={`${prefixCls}-drop-indicator`} />;
}
