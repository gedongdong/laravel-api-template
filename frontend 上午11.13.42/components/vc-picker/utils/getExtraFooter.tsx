import type { PanelMode } from '../interface';

export default function getExtraFooter(
  prefixCls: string,
  mode: PanelMode,
  renderExtraFooter?: (mode: PanelMode) => any,
) {
  if (!renderExtraFooter) {
    return null;
  }

  return <div class={`${prefixCls}-footer-extra`}>{renderExtraFooter(mode)}</div>;
}
