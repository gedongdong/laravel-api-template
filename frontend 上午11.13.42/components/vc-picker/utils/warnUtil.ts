import { warning } from '../../vc-util/warning';
import type { DisabledTimes, PickerMode } from '../interface';

export interface WarningProps extends DisabledTimes {
  picker?: PickerMode;
}

export function legacyPropsWarning(props: WarningProps) {
  const { picker, disabledHours, disabledMinutes, disabledSeconds } = props;

  if (picker === 'time' && (disabledHours || disabledMinutes || disabledSeconds)) {
    warning(
      false,
      `'disabledHours', 'disabledMinutes', 'disabledSeconds' will be removed in the next major version, please use 'disabledTime' instead.`,
    );
  }
}
