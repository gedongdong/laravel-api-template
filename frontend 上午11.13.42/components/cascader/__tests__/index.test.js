import { asyncExpect } from '../../../tests/utils';
import { mount } from '@vue/test-utils';
import KeyCode from '../../_util/KeyCode';
import Cascader from '..';
import focusTest from '../../../tests/shared/focusTest';

function $$(className) {
  return document.body.querySelectorAll(className);
}
const options = [
  {
    value: 'zhejiang',
    label: 'Zhejiang',
    children: [
      {
        value: 'hangzhou',
        label: 'Hangzhou',
        children: [
          {
            value: 'xihu',
            label: 'West Lake',
          },
        ],
      },
    ],
  },
  {
    value: 'jiangsu',
    label: 'Jiangsu',
    children: [
      {
        value: 'nanjing',
        label: 'Nanjing',
        children: [
          {
            value: 'zhonghuamen',
            label: 'Zhong Hua Men',
          },
        ],
      },
    ],
  },
];

function filter(inputValue, path) {
  return path.some(option => option.label.toLowerCase().indexOf(inputValue.toLowerCase()) > -1);
}

function toggleOpen(wrapper) {
  wrapper.find('.ant-select-selector').trigger('mousedown');
}

function isOpen(wrapper) {
  return !!wrapper.findComponent({ name: 'Trigger' }).props().popupVisible;
}

describe('Cascader', () => {
  focusTest(Cascader);
  beforeEach(() => {
    document.body.innerHTML = '';
  });
  it('popup correctly when panel is hidden', async () => {
    mount(Cascader, {
      props: { options },
      sync: false,
      attachTo: 'body',
    });
    await asyncExpect(() => {
      expect($$('.ant-cascader-menus').length).toBe(0);
    });
  });

  it('popup correctly when panel is open', async () => {
    const wrapper = mount(Cascader, { props: { options }, sync: false, attachTo: 'body' });
    await asyncExpect(() => {
      toggleOpen(wrapper);
    });
    expect($$('.ant-cascader-menus').length).toBe(1);
    await asyncExpect(() => {
      expect($$('.ant-cascader-menus')[0].parentNode.parentNode.innerHTML).toMatchSnapshot();
    }, 1000);
  });

  it('support controlled mode', async () => {
    const wrapper = mount(Cascader, { props: { options }, sync: false });
    await asyncExpect(() => {
      wrapper.setProps({
        value: ['zhejiang', 'hangzhou', 'xihu'],
      });
    });
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  it('popup correctly with defaultValue', async () => {
    const wrapper = mount(Cascader, {
      props: {
        options,
        defaultValue: ['zhejiang', 'hangzhou'],
      },
      sync: false,
    });

    await asyncExpect(() => {
      toggleOpen(wrapper);
    });
    expect($$('.ant-cascader-menus').length).toBe(1);
    await asyncExpect(() => {
      expect($$('.ant-cascader-menus')[0].parentNode.parentNode.innerHTML).toMatchSnapshot();
    }, 0);
  });

  it('can be selected', async () => {
    const wrapper = mount(Cascader, { props: { options }, sync: false });
    await asyncExpect(() => {
      toggleOpen(wrapper);
    });
    await asyncExpect(() => {
      $$('.ant-cascader-menu')[0].querySelectorAll('.ant-cascader-menu-item')[0].click();
    });

    await asyncExpect(() => {
      expect($$('.ant-cascader-menus')[0].innerHTML).toMatchSnapshot();
    });

    await asyncExpect(() => {
      $$('.ant-cascader-menu')[1].querySelectorAll('.ant-cascader-menu-item')[0].click();
    });

    await asyncExpect(() => {
      expect($$('.ant-cascader-menus')[0].innerHTML).toMatchSnapshot();
    });

    await asyncExpect(() => {
      $$('.ant-cascader-menu')[2].querySelectorAll('.ant-cascader-menu-item')[0].click();
    });

    await asyncExpect(() => {
      expect($$('.ant-cascader-menus')[0].innerHTML).toMatchSnapshot();
    });
  });

  fit('backspace should work with `Cascader[showSearch]`', async () => {
    const wrapper = mount(Cascader, { props: { options, showSearch: true }, sync: false });
    await asyncExpect(() => {
      wrapper.find('input').element.value = '123';
      wrapper.find('input').trigger('input');
    });
    await asyncExpect(() => {
      expect(isOpen(wrapper)).toBeTruthy();
    });
    await asyncExpect(() => {
      wrapper.find('input').element.keyCode = KeyCode.BACKSPACE;
      wrapper.find('input').trigger('keydown');
    });
    await asyncExpect(() => {
      expect(isOpen(wrapper)).toBeTruthy();
    });
    await asyncExpect(() => {
      wrapper.find('input').element.value = '';
      wrapper.find('input').trigger('input');
    });
    await asyncExpect(() => {
      expect(isOpen(wrapper)).toBeTruthy();
    });
    // await asyncExpect(() => {
    //   wrapper.find('input').element.keyCode = KeyCode.BACKSPACE;
    //   wrapper.find('input').trigger('keydown');
    // });
    // await asyncExpect(() => {
    //   expect(isOpen(wrapper)).toBeFalsy();
    // }, 0);
  });

  describe('limit filtered item count', () => {
    beforeEach(() => {
      document.body.outerHTML = '';
    });

    afterEach(() => {
      document.body.outerHTML = '';
    });

    it('limit with positive number', async () => {
      const wrapper = mount(Cascader, {
        props: { options, showSearch: { filter, limit: 1 } },
        sync: false,
        attachTo: 'body',
      });
      wrapper.find('input').trigger('click');
      wrapper.find('input').element.value = 'a';
      wrapper.find('input').trigger('input');
      await asyncExpect(() => {
        expect($$('.ant-cascader-menu-item').length).toBe(1);
      }, 0);
    });

    it('not limit', async () => {
      const wrapper = mount(Cascader, {
        props: { options, showSearch: { filter, limit: false } },
        sync: false,
        attachTo: 'body',
      });
      wrapper.find('input').trigger('click');
      wrapper.find('input').element.value = 'a';
      wrapper.find('input').trigger('input');
      await asyncExpect(() => {
        expect($$('.ant-cascader-menu-item').length).toBe(2);
      }, 0);
    });

    it('negative limit', async () => {
      const wrapper = mount(Cascader, {
        props: { options, showSearch: { filter, limit: -1 } },
        sync: false,
        attachTo: 'body',
      });
      wrapper.find('input').trigger('click');
      wrapper.find('input').element.value = 'a';
      wrapper.find('input').trigger('input');
      await asyncExpect(() => {
        expect($$('.ant-cascader-menu-item').length).toBe(2);
      }, 0);
    });
  });
});
