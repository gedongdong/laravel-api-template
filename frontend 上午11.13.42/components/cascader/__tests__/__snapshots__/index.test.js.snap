// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cascader can be selected 1`] = `
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Zhejiang" aria-checked="false" data-path-key="zhejiang">
    <!---->
    <div class="ant-cascader-menu-item-content">Zhejiang</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Jiangsu" aria-checked="false" data-path-key="jiangsu">
    <!---->
    <div class="ant-cascader-menu-item-content">Jiangsu</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
</ul>
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Hangzhou" aria-checked="false" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou">
    <!---->
    <div class="ant-cascader-menu-item-content">Hangzhou</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
</ul>
`;

exports[`Cascader can be selected 2`] = `
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Zhejiang" aria-checked="false" data-path-key="zhejiang">
    <!---->
    <div class="ant-cascader-menu-item-content">Zhejiang</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Jiangsu" aria-checked="false" data-path-key="jiangsu">
    <!---->
    <div class="ant-cascader-menu-item-content">Jiangsu</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
</ul>
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Hangzhou" aria-checked="false" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou">
    <!---->
    <div class="ant-cascader-menu-item-content">Hangzhou</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
</ul>
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item" role="menuitemcheckbox" title="West Lake" aria-checked="false" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou__RC_CASCADER_SPLIT__xihu">
    <!---->
    <div class="ant-cascader-menu-item-content">West Lake</div>
    <!---->
    <!---->
  </li>
</ul>
`;

exports[`Cascader can be selected 3`] = `
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Zhejiang" aria-checked="true" data-path-key="zhejiang">
    <!---->
    <div class="ant-cascader-menu-item-content">Zhejiang</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Jiangsu" aria-checked="false" data-path-key="jiangsu">
    <!---->
    <div class="ant-cascader-menu-item-content">Jiangsu</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
</ul>
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Hangzhou" aria-checked="true" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou">
    <!---->
    <div class="ant-cascader-menu-item-content">Hangzhou</div>
    <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
    <!---->
  </li>
</ul>
<ul class="ant-cascader-menu" role="menu">
  <li class="ant-cascader-menu-item ant-cascader-menu-item-active" role="menuitemcheckbox" title="West Lake" aria-checked="true" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou__RC_CASCADER_SPLIT__xihu">
    <!---->
    <div class="ant-cascader-menu-item-content">West Lake</div>
    <!---->
    <!---->
  </li>
</ul>
`;

exports[`Cascader popup correctly when panel is open 1`] = `
<div>
  <div class="ant-cascader-menus">
    <ul class="ant-cascader-menu" role="menu">
      <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Zhejiang" aria-checked="false" data-path-key="zhejiang">
        <!---->
        <div class="ant-cascader-menu-item-content">Zhejiang</div>
        <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
        <!---->
      </li>
      <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Jiangsu" aria-checked="false" data-path-key="jiangsu">
        <!---->
        <div class="ant-cascader-menu-item-content">Jiangsu</div>
        <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
        <!---->
      </li>
    </ul>
  </div>
</div>
`;

exports[`Cascader popup correctly with defaultValue 1`] = `
<div>
  <div class="ant-cascader-menus">
    <ul class="ant-cascader-menu" role="menu">
      <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Zhejiang" aria-checked="true" data-path-key="zhejiang">
        <!---->
        <div class="ant-cascader-menu-item-content">Zhejiang</div>
        <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
        <!---->
      </li>
      <li class="ant-cascader-menu-item ant-cascader-menu-item-expand" role="menuitemcheckbox" title="Jiangsu" aria-checked="false" data-path-key="jiangsu">
        <!---->
        <div class="ant-cascader-menu-item-content">Jiangsu</div>
        <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
        <!---->
      </li>
    </ul>
    <ul class="ant-cascader-menu" role="menu">
      <li class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active" role="menuitemcheckbox" title="Hangzhou" aria-checked="true" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou">
        <!---->
        <div class="ant-cascader-menu-item-content">Hangzhou</div>
        <div class="ant-cascader-menu-item-expand-icon"><span role="img" aria-label="right" class="anticon anticon-right"><svg focusable="false" class="" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg></span></div>
        <!---->
      </li>
    </ul>
    <ul class="ant-cascader-menu" role="menu">
      <li class="ant-cascader-menu-item" role="menuitemcheckbox" title="West Lake" aria-checked="true" data-path-key="zhejiang__RC_CASCADER_SPLIT__hangzhou__RC_CASCADER_SPLIT__xihu">
        <!---->
        <div class="ant-cascader-menu-item-content">West Lake</div>
        <!---->
        <!---->
      </li>
    </ul>
  </div>
</div>
`;

exports[`Cascader support controlled mode 1`] = `
<div class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow">
  <!---->
  <!---->
  <div class="ant-select-selector"><span class="ant-select-selection-search"><input id="rc_select_TEST_OR_SSR" autocomplete="off" class="ant-select-selection-search-input" style="opacity: 0;" role="combobox" aria-haspopup="listbox" aria-owns="rc_select_TEST_OR_SSR_list" aria-autocomplete="list" aria-controls="rc_select_TEST_OR_SSR_list" readonly="" unselectable="on" type="search"></span><span class="ant-select-selection-item" title="Zhejiang">Zhejiang</span>
    <!---->
  </div><span class="ant-select-arrow" style="user-select: none;" unselectable="on" aria-hidden="true"><span role="img" aria-label="down" class="anticon anticon-down ant-select-suffix"><svg focusable="false" class="" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path></svg></span></span><span class="ant-select-clear" style="user-select: none;" unselectable="on" aria-hidden="true"><span role="img" aria-label="close-circle" class="anticon anticon-close-circle"><svg focusable="false" class="" data-icon="close-circle" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"></path></svg></span></span>
</div>
`;
